notebooks/kaggle/store-sales-time-series-forecasting/experiments/linear_regression_v2.ipynb

```diff
@@ -0,0 +1,564 @@
+    "        pdp.OneHotEncode('store_nbr'),\n",
+    "    ])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9bc34049",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "X = transform_pipeline.apply(X)\n",
+    "X.shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1f998de0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "ts_cv = TimeSeriesSplit(\n",
+    "    n_splits=4,\n",
+    "    gap=0,\n",
+    "    max_train_size=365*54,\n",
+    "    test_size=15*54,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b6e295de",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "oil_data = pd.read_csv(DATA_ROOT / 'oil.csv')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "1def044b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "test_data = pd.read_csv(DATA_ROOT / 'test.csv', index_col = 'id')\n",
+    "test_data = test_data.merge(oil_data, on='date', how='left')\n",
+    "test_data['dcoilwtico'] = test_data['dcoilwtico'].fillna(method='ffill')\n",
+    "test_data = test_data.sort_values(by=['date', 'store_nbr'], ascending=True, ignore_index=True)\n",
+    "\n",
+    "test_data_copy = test_data.copy()\n",
+    "\n",
+    "test_data = test_data.drop('date', axis=1)\n",
+    "test_data = transform_pipeline(test_data)\n",
+    "test_data['family'] = test_data['family'].str.lower()\n",
+    "test_data.shape"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d472b858",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "submission = pd.read_csv(DATA_ROOT / 'sample_submission.csv')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0f6e6ba0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def training_algorithm(X, test_data, model=LinearRegression()):\n",
+    "    scores_RMSLE = []\n",
+    "    scores_RMSE = []\n",
+    "    scores_MAE = []\n",
+    "    scores_MAPE = []\n",
+    "    scores_R2 = []\n",
+    "    count = 0\n",
+    "    for family in X['family'].unique():\n",
+    "        X_train = X[X['family'] == family].drop(columns=['sales', 'family'])\n",
+    "        X_train_copy = X_train.copy()\n",
+    "        y_train = X[X['family'] == family]['sales']\n",
+    "        for train_index, val_index in ts_cv.split(X_train_copy):\n",
+    "            X_train_cv, X_val_cv = X_train_copy.iloc[train_index], X_train_copy.iloc[val_index]\n",
+    "            y_train_cv, y_val_cv = y_train.iloc[train_index], y_train.iloc[val_index]\n",
+    "            model_cv = LinearRegression()\n",
+    "            model_cv.fit(X_train_cv, y_train_cv)\n",
+    "            y_val_cv_pred = model_cv.predict(X_val_cv)\n",
+    "            y_val_cv_pred[y_val_cv_pred < 0] = 0\n",
+    "            if (count + 1) % ts_cv.n_splits == 0:\n",
```

Comment:
Not sure why you consider the last split only. You might want also to compute mean + std of your metrics across different splits, please see an [example](https://scikit-learn.org/stable/auto_examples/applications/plot_cyclical_feature_engineering.html#gradient-boosting) here.